File: Hot4/Hot4.Service/Concrete/NetworkClientFactory.cs
using Hot4.Service.Abstract;
using System;
using System.Collections.Generic;
using System.Linq;
using System.ServiceModel;
using System.Text;
using System.Threading.Tasks;

namespace Hot4.Service.Concrete
{
    public class NetworkClientFactory
    {
        public static INetworkClient<T> Create<T>(string endpointAddress, int timeout = 30000)
        {
            var binding = new BasicHttpBinding
            {
                SendTimeout = TimeSpan.FromSeconds(timeout / 1000) // Convert milliseconds to seconds
            };

            // Create the channel factory for the service endpoint
            var channelFactory = new ChannelFactory<T>(binding, new EndpointAddress(endpointAddress));

            // Wrap the factory in a wrapper class and return it
            var wrapper = new ChannelFactoryWrapper<T>(channelFactory);
            return wrapper;
        }
    }
}

File: Customer APIs/hot.backend.all/hot.core/hot.core/ServiceWrappers/NetworkClientFactory.vb
Imports System.ServiceModel

Namespace ServiceWrappers

    Public Class NetworkClientFactory
        Public Shared Function Create(Of T)(endpointAddress As String, Optional timeout As Integer = 30000) As INetworkClient(Of T)
            Dim binding = New BasicHttpBinding()
            binding.SendTimeout = New TimeSpan(0, 0, timeout / 1000)
            Dim channelFactory As ChannelFactory = New ChannelFactory(Of T)(binding, New EndpointAddress(endpointAddress))
            Dim wrapper = New ChannelFactoryWrapper(Of T)(channelFactory)
            Return wrapper
        End Function
    End Class
End NameSpace